### DISCUSSION TITLE
Google Login Problem

### DISCUSSION THREAD

**Discussion by GlenEaston** [Budibase/budibase#7513] [Help]
Hi,
I'm having an issue logging into my google account. I'm getting the following error. I have localhost and localhost:10000 on my account, but I can't put this ip address because google says it needs a domain. What am I doing wrong?

Authorization Error
Error 400: invalid_request

You can't sign in to this app because it doesn't comply with Google's OAuth 2.0 policy for keeping apps secure.

You can let the app developer know that this app doesn't comply with one or more Google validation rules.

redirect_uri: http://xxx.xxx.xxx.xx:10000//api/global/auth/datasource/google/callback

fyi: I'm using digital ocean, and I can otherwise log into BB without issue.

**Comment by melohagan** [comment]
Hi @GlenEaston 

Is there anymore detail on the error? I.e. why is the request invalid? 

Maybe your callback URL should look like this: https://docs.budibase.com/docs/sso-with-google

**Comment by GlenEaston** [comment]
No luck. I think the issue is that the callback needs to be to HTTPS not
HTTP plus a site that is not an IP address (needs a .com, etc). So, that
makes it problematic for test environments...

**Comment by shogunpurple** [comment]
@GlenEaston Google clearly states you cannot use a public IP address as part of their OAuth configuration:

> For use with requests from a web server. This is the path in your application that users are redirected to after they have authenticated with Google. The path will be appended with the authorization code for access. Must have a protocol. Cannot contain URL fragments or relative paths. Cannot be a public IP address.

You can use `localhost` and even the loopback address on `127.0.0.1`, because google knows these are just local.

If you have confirmed that your google login is working locally, the next step to get it working in your digitalocean environment is to set up a custom domain on it, and use that as your redirect URL.

More info on how to set up a custom domain here:
https://docs.budibase.com/docs/reverse-proxy#install-caddy
